kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/MetadataChecksumGenerator.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.zip.CRC32;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+public class MetadataChecksumGenerator {
+
+    public static final String REFERENT_CHECKSUM_ANNOTATION = "kroxylicious.io/referent-checksum";
+
+    private MetadataChecksumGenerator() {
+    }
+
+    public static String checksumFor(HasMetadata... metadataSources) {
```

Comment:
I suspect the `...` will get awkward when you have to detail with things like the VKC depending on a variable number of filters and ingresses. You'll probably end up changing it to a `List`.